plugin/storage/grpc/config/config.go

```diff
@@ -47,10 +50,11 @@ type Configuration struct {
 	RemoteConnectTimeout    time.Duration `yaml:"connection-timeout" mapstructure:"connection-timeout"`
 	TenancyOpts             tenancy.Options
 
-	pluginHealthCheck     *time.Ticker
-	pluginHealthCheckDone chan bool
-	pluginRPCClient       plugin.ClientProtocol
-	remoteConn            *grpc.ClientConn
+	pluginHealthCheck       *time.Ticker
+	pluginHealthCheckDone   chan bool
+	pluginRPCClient         plugin.ClientProtocol
+	remoteConn              *grpc.ClientConn
+	configgrpc.ClientConfig `mapstructure:",squash"`
```

Comment:
Let's create a new config struct like ConfigV2 and only keep the fields that are really needed there. For instance, we do not plan to support sidecar plugin model, so the first three fields at the top are not needed.

Also, I assume otel ClientConfig would already include things like TLS and timeout and address, so fields 4-6 also should not be needed in the new config.

Basically, I would suggest this refactoring:
  * add ConfigV2 with only what we need
  * refactor remote storage initialization code to use OTEL helpers for interpreting ClientConfig into a gRPC connection
  * write a translator from v1 Configuration into ConfigV2 (V2 should be a superset, except for the sidecar mode).

Reply:
now the sidecar plugin config fields are removed, do we still need to create a new configV2 and a translator?